Azure-Functions crash when X-Forwarded headers are enabled java.lang.NullPointerException

### Describe the bug

I'm trying to deploy a quarkus application to azure functions behind a proxy.

After I added those two properties `quarkus.http.proxy.proxy-address-forwarding=true` and `quarkus.http.proxy.enable-forwarded-host=true` the application started to crash whenever it receives a request.


### Expected behavior

The application should not crash. It should take into account X-Forwarded headers when it receives them by creating absolute urls with values provided in the headers. Additionally, it should still work if those headers are not provided in the request.

### Actual behavior

Quarkus crashes after receiving a request when the application is deployed to azure functions with those parameters enabled:
quarkus.http.proxy.proxy-address-forwarding=true
quarkus.http.proxy.enable-forwarded-host=true

It doesn't seem to matter if the X-Forwarded headers are part of the request or not.

```
2023-09-04T07:17:53Z   [Information]   jakarta.ws.rs.WebApplicationException: HTTP 400 Bad Request
2023-09-04T07:17:53Z   [Information]   	at org.jboss.resteasy.reactive.server.handlers.ParameterHandler.handle(ParameterHandler.java:66)
2023-09-04T07:17:53Z   [Information]   	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:137)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
2023-09-04T07:17:53Z   [Information]   	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:577)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
2023-09-04T07:17:53Z   [Information]   	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2023-09-04T07:17:53Z   [Information]   	at java.base/java.lang.Thread.run(Thread.java:833)
2023-09-04T07:17:53Z   [Information]   Caused by: java.lang.NullPointerException: Cannot invoke "io.vertx.core.net.SocketAddress.port()" because "this.remoteAddress" is null
2023-09-04T07:17:53Z   [Information]   	at io.quarkus.vertx.http.runtime.ForwardedParser.calculate(ForwardedParser.java:170)
2023-09-04T07:17:53Z   [Information]   	at io.quarkus.vertx.http.runtime.ForwardedParser.uri(ForwardedParser.java:105)
2023-09-04T07:17:53Z   [Information]   	at io.quarkus.vertx.http.runtime.ForwardedServerRequestWrapper.uri(ForwardedServerRequestWrapper.java:138)
2023-09-04T07:17:53Z   [Information]   	at io.vertx.ext.web.impl.HttpServerRequestWrapper.uri(HttpServerRequestWrapper.java:79)
2023-09-04T07:17:53Z   [Information]   	at io.vertx.ext.web.impl.RoutingContextImpl.getQueryParams(RoutingContextImpl.java:465)
2023-09-04T07:17:53Z   [Information]   	at io.vertx.ext.web.impl.RoutingContextImpl.queryParams(RoutingContextImpl.java:445)
2023-09-04T07:17:53Z   [Information]   	at io.vertx.ext.web.impl.RoutingContextWrapper.queryParams(RoutingContextWrapper.java:318)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.resteasy.reactive.server.vertx.VertxResteasyReactiveRequestContext.getQueryParam(VertxResteasyReactiveRequestContext.java:204)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext.getQueryParameter(ResteasyReactiveRequestContext.java:824)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.resteasy.reactive.server.core.parameters.QueryParamExtractor.extractParameter(QueryParamExtractor.java:22)
2023-09-04T07:17:53Z   [Information]   	at org.jboss.resteasy.reactive.server.handlers.ParameterHandler.handle(ParameterHandler.java:45)
2023-09-04T07:17:53Z   [Information]   	... 9 more
```

### How to Reproduce?

1. create a simple quarkus application with QuarkusResteasyReactive dependency
2. enable azure-functions support by adding those properties
```
quarkus.http.root-path=/api
quarkus.http.port=${FUNCTIONS_HTTPWORKER_PORT}
```
and this dependency
```
quarkus-azure-functions-http
```
3. build the application `quarkus build`
4. zip content of the target/azure-function directory
5. create azure function in the azure cloud
6. deploy the zip file with `az functionapp deployment source config-zip -g your_resource_group -n your-azure-application-name --src ./your-zip-file.zip`
7. open the log stream at azure
8. send a request

### Output of `uname -a` or `ver`

Linux 55ea7e5de999 5.15.116.1-1.cm2 #1 SMP Wed Jun 21 23:17:09 UTC 2023 x86_64 GNU/Linux

### Output of `java -version`

openjdk 17.0.6 2023-01-17 LTS OpenJDK Runtime Environment Microsoft-7209853 (build 17.0.6+10-LTS) OpenJDK 64-Bit Server VM Microsoft-7209853 (build 17.0.6+10-LTS, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.3.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: /home/my-username/apache-maven-3.9.2 Java version: 17.0.7, vendor: Eclipse Adoptium, runtime: /opt/mandrel-java17-22.3.2.1-Final Default locale: en_US, platform encoding: UTF-8 OS name: "linux", version: "6.2.0-31-generic", arch: "amd64", family: "unix"

### Additional information

_No response_

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedParser.java
@@ -140,7 +140,7 @@ class ForwardedParser {
 
                 matcher = FORWARDED_FOR_PATTERN.matcher(forwarded);
                 if (matcher.find()) {
-                    remoteAddress = parseFor(matcher.group(1).trim(), remoteAddress.port());
+                    remoteAddress = parseFor(matcher.group(1).trim(), remoteAddress != null ? remoteAddress.port() : port);
                 }
             } else if (forwardingProxyOptions.allowXForwarded) {
                 String protocolHeader = delegate.getHeader(X_FORWARDED_PROTO);
@@ -177,7 +177,7 @@ class ForwardedParser {
 
                 String forHeader = delegate.getHeader(X_FORWARDED_FOR);
                 if (forHeader != null) {
-                    remoteAddress = parseFor(getFirstElement(forHeader), remoteAddress.port());
+                    remoteAddress = parseFor(getFirstElement(forHeader), remoteAddress != null ? remoteAddress.port() : port);
                 }
             }
         }

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedProxyHandler.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/ForwardedProxyHandler.java
@@ -45,7 +45,11 @@ public class ForwardedProxyHandler implements Handler<HttpServerRequest> {
 
     @Override
     public void handle(HttpServerRequest event) {
-        if (event.remoteAddress().isDomainSocket()) {
+        if (event.remoteAddress() == null) {
+            // client address may not be available with virtual http channel
+            LOGGER.debug("Client address is not available, 'Forwarded' and 'X-Forwarded' headers are going to be ignored");
+            handleForwardedServerRequest(event, denyAll());
+        } else if (event.remoteAddress().isDomainSocket()) {
             // we do not support domain socket proxy checks, ignore the headers
             LOGGER.debug("Domain socket are not supported, 'Forwarded' and 'X-Forwarded' headers are going to be ignored");
             handleForwardedServerRequest(event, denyAll());

